### PR TITLE
boards: arm: add nrf9131ek_nrf9131

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch adds the nRF9131-EK board.

### CODE REVIEW

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+		slot1_partition: partition@80000 {
+			label = "image-1";
+		};
+		slot1_ns_partition: partition@c0000 {
+			label = "image-1-nonsecure";
+		};
+		scratch_partition: partition@f0000 {
+			label = "image-scratch";
+			reg = <0x000f0000 0xa000>;
+		};
```
Comment (nordicjm) [inline boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi:195]: really should not be using scratch in 2023

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+identifier: nrf9131ek_nrf9131
+name: nRF9131-EK-NRF9131
+type: mcu
+arch: arm
+toolchain:
+  - gnuarmemb
+  - xtools
+  - zephyr
+ram: 88
```
Comment (nordicjm) [inline boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml:9]: 88KB of RAM?

Comment (maxd-nordic) [reply boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml:9]: should I put 128kB of usable ram (for non-secure) or 256kB of total SoC ram?

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi (hunk 1 above)
Comment (maxd-nordic) [reply boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi]: fixed

**Comment by maxd-nordic** [comment]
@nordicjm these things were copied from nrf9161dk. should they be fixed there as well?

**Comment by maxd-nordic** [comment]
For non-secure builds, this one might be needed: https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/97

**Comment by nordicjm** [comment]
> @nordicjm these things were copied from nrf9161dk. should they be fixed there as well?

Yes

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml (hunk 2 above)
Comment (nordicjm) [reply boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml]: 256 probably since tests can use the full 256KiB (I presume?)

File: boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ toolchain:
   - gnuarmemb
   - xtools
   - zephyr
-ram: 88
+ram: 256
 flash: 1024
```
Comment (nordicjm) [inline boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161.yaml:10]: Scratch partition should be removed from 9161 also (can do it in a different PR)

**Review by nordicjm — DISMISSED** [review]

**Review by joerchan — DISMISSED** [review]
TF-M Integration looks good, ran the integration tests and they are passing.

File: boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161.yaml (hunk 3 above)
Comment (maxd-nordic) [reply boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161.yaml:10]: just added the change for the nrf9160dk as well :+1:

**Comment by nordicjm** [comment]
```
		slot1_ns_partition: partition@c5000 {
			reg = <0x000c0000 0x35000>;
```
These 2 conflict

**Comment by maxd-nordic** [comment]
> ```
> 		slot1_ns_partition: partition@c5000 {
> 			reg = <0x000c0000 0x35000>;
> ```
> 
> These 2 conflict

Sorry! Clumsy editing here. Maybe these partition configs should be in one file?

**Comment by carlescufi** [comment]
@anangl can you take a look?

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+&uart0 {
+	status = "okay";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart0_default>;
+	pinctrl-1 = <&uart0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+arduino_serial: &uart1 {
```
Comment (anangl) [inline boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi:92]: Does this board actually contain the Arduino header (there is no related nexus node definition in this file)?

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi [hunk 5]
```diff
@@ -156,6 +156,9 @@ arduino_spi: &spi3 {
 		reg = <0>;
 		spi-max-frequency = <8000000>;
 		size = <268435456>;
+		has-dpd;
+		t-enter-dpd = <3000>;
+		t-exit-dpd = <40000>;
```
Comment (anangl) [inline boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi:161]: This should be moved to the first commit ("boards: arm: add nrf9131ek_nrf9131").

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_ns.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+identifier: nrf9131ek_nrf9131_ns
+name: nRF9131-EK-NRF9131-Non-Secure
+type: mcu
+arch: arm
+toolchain:
+  - gnuarmemb
+  - xtools
+  - zephyr
+ram: 128
+flash: 276
```
Comment (anangl) [inline boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_ns.yaml:10]: Is this correct? `slot0_ns_partition` size is 212 kB (0x35000).

File: boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161.yaml [hunk 7]
```diff
@@ -6,7 +6,7 @@ toolchain:
   - gnuarmemb
   - xtools
   - zephyr
-ram: 88
+ram: 256
```
Comment (anangl) [inline boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161.yaml:9]: I believe this should not be changed. `zephyr,sram` points to `sram0_s` and the size of this node is `DT_SIZE_K(88)`, so that's the RAM size available for samples and tests.

**Comment by joerchan** [comment]
@carlescufi I'd like https://github.com/zephyrproject-rtos/zephyr/pull/63515 to go in first. That way we don't have to modify TF-M with the changes in https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/97 in order to get the NS board working.

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi (hunk 4 above)
Comment (maxd-nordic) [reply boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi:92]: It doesn't. I'll remove the arduino labels.

**Comment by joerchan** [comment]
@maxd-nordic Did you rebase this PR and check that the NS board works?

**Comment by maxd-nordic** [comment]
> @maxd-nordic Did you rebase this PR and check that the NS board works?

Just checked - all tests pass! :)

**Review by mbolivar-ampere — COMMENTED** [review]
ACK on the tests/lib/devicetree/devices/testcase.yaml changes

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_ns_defconfig [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+CONFIG_PINCTRL=y
+
+# enable PMIC
+CONFIG_I2C=y
+CONFIG_REGULATOR=y
+CONFIG_SENSOR=y
+CONFIG_NPM1300_CHARGER=y
```
Comment (nordicjm) [inline boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_ns_defconfig:35]: Why are these needed to be on by default for _ns builds?

Comment (maxd-nordic) [reply boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_ns_defconfig:35]: this is because of the charger function - it has to be configured for the hardware to work

Comment (maxd-nordic) [reply boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_ns_defconfig:35]: Because the board will brown out without that configuration. We cannot safely run firmware without the PMIC configured.

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+identifier: nrf9131ek_nrf9131
+name: nRF9131-EK-NRF9131
+type: mcu
+arch: arm
+toolchain:
+  - gnuarmemb
+  - xtools
+  - zephyr
+ram: 256
```
Comment (anangl) [inline boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml:9]: This should match the size of the node chosen as `zephyr,sram`, which is `sram0_s` in this case, so it should be 88 kB.

File: boards/arm/nrf9131ek_nrf9131/Kconfig.defconfig [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+if BUILD_WITH_TFM
+
+# By default, if we build with TF-M, instruct build system to
+# flash the combined TF-M (Secure) & Zephyr (Non Secure) image
+config TFM_FLASH_MERGED_BINARY
+	bool
+	default y
+
+endif # BUILD_WITH_TFM
+
```
Comment (anangl) [inline boards/arm/nrf9131ek_nrf9131/Kconfig.defconfig:25]: This should be removed after #63515 is merged.

File: boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi [hunk 11]
```diff
@@ -230,16 +230,12 @@ arduino_spi: &spi3 {
...
+		slot1_ns_partition: partition@c5000 {
 			label = "image-1-nonsecure";
 		};
-		scratch_partition: partition@f0000 {
-			label = "image-scratch";
-			reg = <0x000f0000 0xa000>;
-		};
 		storage_partition: partition@fa000 {
 			label = "storage";
 			reg = <0x000fa000 0x00006000>;
```
Comment (anangl) [inline boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi:241]: Shouldn't this be done like in dc431a7397a629c9fb7e3a27a803d11bf1103389? @joerchan

File: boards/arm/nrf9131ek_nrf9131/board.cmake [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+if(CONFIG_BOARD_NRF9131EK_NRF9131_NS)
+  set(TFM_PUBLIC_KEY_FORMAT "full")
+endif()
+
+if(CONFIG_TFM_FLASH_MERGED_BINARY)
+  set_property(TARGET runners_yaml_props_target PROPERTY hex_file tfm_merged.hex)
+endif()
+
+# TODO: change to nRF9131_xxAA when such device is available in JLink
+board_runner_args(jlink "--device=nRF9160_xxAA" "--speed=4000")
```
Comment (anangl) [inline boards/arm/nrf9131ek_nrf9131/board.cmake:12]: It looks like support for nRF9131 and nRF9161 was added in [v7.92m](https://www.segger.com/downloads/jlink/ReleaseNotes_JLink.html).

File: boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi (hunk 11 above)
Comment (maxd-nordic) [reply boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi:241]: I suppose this depends on how you want to use the freed up space - I put it in the app image.

File: boards/arm/nrf9131ek_nrf9131/Kconfig.defconfig (hunk 10 above)
Comment (maxd-nordic) [reply boards/arm/nrf9131ek_nrf9131/Kconfig.defconfig]: done

File: boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml (hunk 9 above)
Comment (maxd-nordic) [reply boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml]: done

**Review by anangl — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

File: boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi (hunk 11 above)
Comment (joerchan) [reply boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi:241]: As long as the TF-M partitions are placed correctly, they don't follow the configuration you put here, they are statically defined.

Comment (maxd-nordic) [reply boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi:241]: Is this documented somewhere? It seems a bit hard to understand which partition config ends up being used if parts of the configuration are overwritten by build system magic.

Comment (joerchan) [reply boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi:241]: No. It's not overwritten, it's ignored ;) However in NCS the partition manager symbols are used. So this is only a zephyr restriction.

**Comment by kartben** [comment]
Hey @carlescufi @maxd-nordic, this board has no README... Please add at your earliest convenience.

**Comment by kartben** [comment]
I realized that the board doesn't seem to be publicly available yet. While that's not necessarily an issue per se, it still must have documentation in the form of a README in the tree (see https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#contributing-your-board)

**Comment by kartben** [comment]
> Hey @anangl @maxd-nordic @carlescufi, this board has no README... Please add at your earliest convenience.

Ping - just want to make sure you guys saw this. Thanks!

**Comment by maxd-nordic** [comment]
> > Hey @anangl @maxd-nordic @carlescufi, this board has no README... Please add at your earliest convenience.
> 
> Ping - just want to make sure you guys saw this. Thanks!

Yes, I got your request.
